h/presenters/user_json.py

```diff
@@ -21,8 +15,19 @@ def __init__(self, user):
     def asdict(self):
         return {
             "authority": self.user.authority,
-            "email": self.user.email,
             "userid": self.user.userid,
             "username": self.user.username,
             "display_name": self.user.display_name,
         }
+
+
+class TrustedUserJSONPresenter(UserJSONPresenter):
+    """
+    Present a user's information to a trusted consumer. Add additional
+    metadata that is not public-facing.
+    """
+
+    def asdict(self):
+        user_presented = super(TrustedUserJSONPresenter, self).asdict()
+        user_presented["email"] = self.user.email
```

Comment:
Would composition rather than inheritance be potentially less hairy here?

I'm just thinking that it doesn't couple the two classes together quite as much, for example if an `__init__()` were added to `TrustedUserJSONPresenter()` in future it might forget to call `UserJSONPresenter.__init__()`, or if `UserJSONPresenter.__init__()`'s signature was changed it might cause confusing bugs. People adding things to `UserJSONPresenter` might not realise that they automatically become public API of `TrustedUserJSONPresenter` as well.

Reply:
Yep, I'll do this—we do have an established preference for composition over inheritance. I thought you might flag this! Shoulda just done it that was at the outset.